02.calendar/cal.rb

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+require 'date'
+
+TARGET_MONTH_BLANK_SPACE = 15
+DATE_BLANK_SPACE = 2
+
+options = {}
+
+opt = ARGV.getopts('m:', 'y:')
+options[:month] = opt['m'].to_i if opt['m']
+options[:year] = opt['y'].to_i if opt['y']
+options[:month] = opt['m']&.to_i || Date.today.month
+options[:year] = opt['y']&.to_i || Date.today.year
```

Comment:
```suggestion
options[:month] = opt['m']&.to_i || Date.today.month
options[:year] = opt['y']&.to_i || Date.today.year
```

この2行だけで大丈夫です。

- `a&.b`はaがnilならメソッドbを呼び出したときにnilを返す
- `a || b`はaがnilならbを返す

という言語仕様を活用しています。
プロを目指す人のためのRuby入門をもし持っていたらチェックしてみてください〜。

あと、ハッシュに入れずに普通の変数にした方がシンプルですね。

```suggestion
month = opt['m']&.to_i || Date.today.month
year = opt['y']&.to_i || Date.today.year
```

Reply:
ありがとうございます！ 

> a&.bはaがnilならメソッドbを呼び出したときにnilを返す
a || bはaがnilならbを返す
→この2つを組み合わせるとこんなにシンプルにできるんですね。プロを目指す人のためのRuby入門は次のプラクティスで読むことになるのでしっかりと学んでいこうと思います。

> あと、ハッシュに入れずに普通の変数にした方がシンプルですね。
→普通の変数を用いる形に変更しました。あわせて、不要になった `options = {}'も削除しました。